app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/importfiles/ImportFilesScreen.kt

```diff
@@ -427,13 +403,19 @@ private fun SendButton(
         modifier = modifier,
         title = stringResource(R.string.transferSendButton),
         style = ButtonType.PRIMARY,
-        showIndeterminateProgress = { integrityCheckResult() == AppIntegrityResult.Ongoing || isTransferStarted() },
-        enabled = { importedFiles().isNotEmpty() && !isImporting && isSenderEmailCorrect && !isTransferStarted() },
+        showIndeterminateProgress = { sendStatus() == SendStatus.Pending },
+        enabled = { importedFiles().isNotEmpty() && !isImporting && isSenderEmailCorrect && sendStatus().canEnableButton() },
         progress = progress,
         onClick = { sendTransfer() },
     )
 }
 
+private fun SendStatus.canEnableButton(): Boolean = when (this) {
+    SendStatus.Default,
+    SendStatus.Refused -> true
```

Comment:
If the state is Failure, shouldn't the button should be re-enabled too?

Reply:
If the state is failure or success, we're navigating elsewhere, so thought it'd be safer to disable the button to avoid clicking on it once more or something